mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/DeletedMessageVaultContract.java

```diff
@@ -287,9 +291,11 @@ default void deleteExpiredMessagesTaskShouldDeleteOldMails() throws InterruptedE
     @Test
     default void deleteExpiredMessagesTaskShouldDeleteOldMailsWhenRunSeveralTime() throws InterruptedException {
         Mono.from(getVault().append(USER, OLD_DELETED_MESSAGE, new ByteArrayInputStream(CONTENT))).block();
+        getClock().setInstant(NOW.plusYears(4).toInstant());
```

Comment:
Indeed... Got it. Because when we append the 2nd message it's registered on the clock NOW + 4, and we set again the clock to that time, so it's not expired and the retention doesn't apply. Fixing...

Reply:
FYI that was causing the issue in the build reported by @rouazana